### PR TITLE
Add directly used dependencies to pom.xml

### PR DESCRIPTION
**Pull request by lemoncurry**
- [x] the code follows the [Clean Code](https://clean-code-developer.com/) guidelines.
- [x] the necessary tests are either created or updated.
- [x] all status checks (Travis CI, SonarCloud, etc.) pass.
- [x] your commit history is cleaned up.
- [ ] <s>you updated the changelog.</s>
- [ ] <s>you updated necessary documentation within [retest/docs](https://github.com/retest/docs).</s>

## Description

> Add directly used dependencies to pom.xml 

and remove declared unused dependencies

### State of this PR
WIP

### Additional Context
http://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Transitive_Dependencies

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -102,20 +125,22 @@
...
+			<scope>runtime</scope>
 		</dependency>
 
 		<!-- test dependencies -->
+
 		<dependency>
 			<groupId>org.junit.jupiter</groupId>
-			<artifactId>junit-jupiter-engine</artifactId>
-			<version>${junit.jupiter.version}</version>
+			<artifactId>junit-jupiter-api</artifactId>
```
Comment (beatngu13) [inline pom.xml]: You replaced the engine with the API, have you checked the build if the corresponding tests are still being executed?

I recommend to use the include-all dependency: `junit-jupiter`.

File: pom.xml [hunk 2]
```diff
@@ -102,20 +125,22 @@
...
+			<version>5.6.0</version>
 			<scope>test</scope>
 		</dependency>
 
 		<dependency>
-			<groupId>org.junit.vintage</groupId>
-			<artifactId>junit-vintage-engine</artifactId>
-			<version>${junit.jupiter.version}</version>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
```
Comment (beatngu13) [inline pom.xml]: Here you also removed the (vintage) engine. According to the [JUnit 5 migration sample](https://github.com/junit-team/junit5-samples/tree/master/junit5-migration-maven/), one should keep it—but I haven't tested this.

File: pom.xml (hunk 1 above)
Comment (lemoncurry) [reply pom.xml]: @beatngu13 I am discussing all of this together with @githubert - please view this PR rather as a draft PR, I will change the status accordingly

File: pom.xml (hunk 2 above)
Comment (lemoncurry) [reply pom.xml]: @beatngu13 I am discussing all of this together with @githubert - please view this PR rather as a draft PR, I will change the status accordingly

File: pom.xml (hunk 1 above)
Comment (beatngu13) [reply pom.xml]: OK, thx. Happened to me once: https://github.com/retest/recheck/issues/186/

**Pull request merged**